binary_search_tree/tree.py

```diff
@@ -14,44 +18,179 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log(n)) if the tree is balanced
+    # Space Complexity: O(1) (only one node is being added)
+    def add_helper(self, current_root, new_node):
+        if current_root == None:
+            return new_node
+
+        if new_node.key < current_root.key:
+            current_root.left = self.add_helper(current_root.left, new_node)
+        else:
+            current_root.right = self.add_helper(current_root.right, new_node)
+
+        return current_root
+    
     def add(self, key, value = None):
-        pass
+        if self.root == None:
+            self.root = TreeNode(key, value)
+            return self.root
 
-    # Time Complexity: 
-    # Space Complexity: 
+        new_node = TreeNode(key, value)
+        self.add_helper(self.root, new_node)
+
+    # def add(self, key, value = None):
```

Comment:
Generally if you have big blocks of commented code like this, you can remove them unless there's a particular reason to keep them. In this case, this doesn't seem like the case, so you can remove this.

Reply:
There's a few other commented code you might could clean up in the rest of the file, but that's not really the most important thing, so I'll skip pointing out every one. Just keep in mind that if a comment isn't helpful to explaining how code works, you should just get rid of it!